core/src/main/scala/org/apache/spark/sql/delta/commands/alterDeltaTableCommands.scala

```diff
@@ -219,15 +219,19 @@ case class AlterTableAddColumnsDeltaCommand(
       val resolver = sparkSession.sessionState.conf.resolver
       val newSchema = colsToAddWithPosition.foldLeft(oldSchema) {
         case (schema, QualifiedColTypeWithPosition(columnPath, column, None)) =>
-          val (parentPosition, lastSize) =
-            SchemaUtils.findColumnPosition(columnPath, schema, resolver)
+          val parentPosition = SchemaUtils.findColumnPosition(columnPath, schema, resolver)
+          val lastSize = SchemaUtils.getNestedTypeFromPosition(schema, parentPosition) match {
```

Comment:
While we're here, how about calling this insertionPosition? Since that's what the value actually represents.